Northstar.Client/mod/scripts/vscripts/ui/menu_lobby.nut

```diff
@@ -226,6 +228,57 @@ void function InitLobbyMenu()
 	RegisterSignal( "LeaveParty" )
 }
 
+void function ShowToggleProgressionDialog( var button )
+{
+	bool enabled = Progression_GetPreference()
+
+	DialogData dialogData
+	dialogData.menu = GetMenu( "AnnouncementDialog" )
+	dialogData.header = enabled ? "#PROGRESSION_TOGGLE_ENABLED_HEADER" : "#PROGRESSION_TOGGLE_DISABLED_HEADER"
+	dialogData.message = enabled ? "#PROGRESSION_TOGGLE_ENABLED_BODY" : "#PROGRESSION_TOGGLE_DISABLED_BODY"
+	dialogData.image = $"ui/menu/common/dialog_announcement_1"
+
+	AddDialogButton( dialogData, "#NO" )
+	AddDialogButton( dialogData, "#YES", enabled ? DisableProgression : EnableProgression )
+
+	OpenDialog( dialogData )
+}
+
+void function EnableProgression()
+{
+	Progression_SetPreference( true )
+
+	// update the cache just in case something changed
+	UpdateCachedLoadouts_Delayed()
+
+	DialogData dialogData
+	dialogData.menu = GetMenu( "AnnouncementDialog" )
+	dialogData.header = "#PROGRESSION_ENABLED_HEADER"
+	dialogData.message = "#PROGRESSION_ENABLED_BODY"
+	dialogData.image = $"ui/menu/common/dialog_announcement_1"
+
+	AddDialogButton( dialogData, "#OK" )
+
+	EmitUISound( "UI_Menu_Item_Purchased_Stinger" )
+
+	OpenDialog( dialogData )
+}
+
+void function DisableProgression()
+{
+	Progression_SetPreference( false )
+
+	DialogData dialogData
+	dialogData.menu = GetMenu( "AnnouncementDialog" )
+	dialogData.header = "#PROGRESSION_DISABLED_HEADER"
+	dialogData.message = "#PROGRESSION_DISABLED_BODY"
+	dialogData.image = $"ui/menu/common/dialog_announcement_1"
+
+	AddDialogButton( dialogData, "#OK" )
+	
+	OpenDialog( dialogData )
+}
```

Comment:
no sound like when enabling?

Reply:
I couldn't really find a "disable" sound that isn't used for error sounds. (I also want to encourage enabling more than disabling so a little audio thing is nice)